### PR TITLE
feat: support conditional / null carousel children

### LINKED ISSUE #24: Allow "nullish" children
# Issue

While using react-carousel, I found a problem trying to render conditional children:

```js
<Carousel>
    { condition && <div>hello</div> }
    <div>world</div>
</Carousel>
```

I found the problem to be related with the render method of the Carousel:
```js
 {React.Children.map(children, (child, i) => (
     React.cloneElement(child, {
         ...child.props,
         ...
    })
))}
```

We are trying to access the props of a child that is nullish.

# Solution

A solution to this problem would be to use the `toArray` method provided by `React.Children` that will filter out null children.

Like this:
```js
 {React.Children.toArray(children).map((child, i) => (
     React.cloneElement(child, {
         ...child.props,
         ...
    })
))}
```

# Note
I tried to commit the change to a branch but I seem to have no permissions.

Thanks 🙏 
@satazor @jgradim

### PR DESCRIPTION
**Pull request by jgradim**
Closes #24 

Filter out falsy children nodes before rendering

### CODE REVIEW

File: src/Carousel.js [hunk 1]
```diff
@@ -160,7 +162,7 @@ class Carousel extends Component {
                         onTouchStart={ this.handleSliderTouchStart }
                         onTouchMove={ this.handleSliderTouchMove }
                         onTouchEnd={ this.handleSliderTouchEnd } >
-                        {React.Children.map(children, (child, i) => (
+                        {nonNullChildren.map((child, i) => (
```
Comment (satazor) [inline src/Carousel.js]:
```suggestion
                        { nonNullChildren.map((child, i) => (
```

File: demo/pages/index.js [hunk 2]
```diff
@@ -214,6 +214,7 @@ const Home = () => (
         <PropSetter>
             {(props) => (
                 <Carousel { ...props }>
+                    {false && <div />}
```
Comment (satazor) [inline demo/pages/index.js]:
```suggestion
                    {false && <div />}
```

Leftover?

File: src/Carousel.js (hunk 1 above)
Comment (satazor) [reply src/Carousel.js]: I'm surprised this isn't throwing a eslint error. It should. Are there other places like this?

File: demo/pages/index.js (hunk 2 above)
Comment (jgradim) [reply demo/pages/index.js]: Leftover, removing!

**Review by satazor — APPROVED** [review]

File: src/Carousel.js (hunk 1 above)
Comment (jgradim) [reply src/Carousel.js]: I don't see any. Could it be because it's a render interpolation and not a prop? 🤔

**Review by PedroMiguelSS — APPROVED** [review]

File: src/Carousel.js [hunk 3]
```diff
@@ -134,6 +134,8 @@ class Carousel extends Component {
         const { current, dragging } = this.state;
         const offset = this.offset();
 
+        const nonNullChildren = React.Children.toArray(children).filter((c) => c);
```
Comment (paulobmarcos) [inline src/Carousel.js]: When I tested, `React.Children.toArray(children)` seemed to do the job of removing null children. 

Will the filter help with other potential cases?

Comment (jgradim) [reply src/Carousel.js]: I was unable to use just `toArray`, still had the issue of trying to access `props` of `undefined`. Let me check if I did something wrong.

Comment (jgradim) [reply src/Carousel.js]: Correct, `filter` is not needed here. This is an undocumented side effect of `toArray` method, though. I guess it makes sense if you consider how React works.

**Review by satazor — APPROVED** [review]

**Pull request merged**